src/App.tsx

```diff
@@ -2,73 +1,109 @@
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputValue, setInputValue] = useState('');
+  const [filteredPeople, setFilteredPeople] = useState(peopleFromServer);
+  const [selectedPerson, setSelectedPerson] = useState<null | { name: string; born: number; died: number }>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [debounceTimer, setDebounceTimer] = useState<NodeJS.Timeout | null>(null);
+
+  const debounceDelay = 300;
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+
+    setInputValue(value);
+    setSelectedPerson(null);
+    setIsDropdownVisible(true);
+
+    if (debounceTimer) {
+      clearTimeout(debounceTimer);
+    }
+
+    const newTimer = setTimeout(() => {
+      if (value === '') {
+        setFilteredPeople(peopleFromServer);
+      } else {
+        setFilteredPeople(
+          peopleFromServer.filter(person =>
+            person.name.toLowerCase().includes(value.toLowerCase())
+          )
+        );
+      }
+    }, debounceDelay);
+
+    setDebounceTimer(newTimer);
+  };
+
+  const handleSelectPerson = (person: { name: string; born: number; died: number }) => {
+    setSelectedPerson(person);
+    setInputValue(person.name);
+    setIsDropdownVisible(false);
+  };
+
+  const handleClickOutside = (e: MouseEvent) => {
+    const target = e.target as HTMLElement;
+    if (!target.closest('.autocomplete')) {
```

Comment:
The `handleClickOutside` function uses `closest` to check if the click was outside the autocomplete. Ensure that the `.autocomplete` class is applied correctly to the relevant elements to avoid unexpected behavior. Additionally, consider using `useRef` to directly reference the dropdown element for more precise control.